src/ImageSharp/Formats/Jpeg/Components/Decoder/HuffmanTable.cs

```diff
@@ -62,20 +55,7 @@ internal unsafe struct HuffmanTable
         /// <param name="values">The huffman values</param>
         public HuffmanTable(ReadOnlySpan<byte> codeLengths, ReadOnlySpan<byte> values)
         {
-            this.isConfigured = false;
-            Unsafe.CopyBlockUnaligned(ref this.Sizes[0], ref MemoryMarshal.GetReference(codeLengths), (uint)codeLengths.Length);
-            Unsafe.CopyBlockUnaligned(ref this.Values[0], ref MemoryMarshal.GetReference(values), (uint)values.Length);
-        }
-
-        /// <summary>
-        /// Expands the HuffmanTable into its readable form.
-        /// </summary>
-        public void Configure()
-        {
-            if (this.isConfigured)
-            {
-                return;
-            }
+            Unsafe.CopyBlockUnaligned(ref this.values[0], ref MemoryMarshal.GetReference(values), (uint)values.Length);
 
             Span<char> huffSize = stackalloc char[257];
             Span<uint> huffCode = stackalloc uint[257];
```

Comment:
@antonfirsov this concerns me a bit, are we okay allocating 1.5kb on stack here? This part was taken from current master implementation.

Reply:
Hmmm yeah. Never did the maths on that previously. Anything larger than a kilobyte is generally considered a concern.

Reply:
It would look ugly but we are already pooling some memory for table parsing:

https://github.com/SixLabors/ImageSharp/blob/a54a19fa994ebed83f3c76a30482c96618a8835d/src/ImageSharp/Formats/Jpeg/JpegDecoderCore.cs#L1142-L1144

We can pool those 256 bytes + 1.5kb in one package and split it into 3 different spans for table building workspace.

Reply:
Marked this as a draft until I'll implement this.